api/src/apps/routes.rs

```diff
@@ -513,4 +513,522 @@ mod tests {
+            let get = client
+                .get(rocket::uri!(crate::apps::routes::apps))
+                .header(Header::new("x-forwarded-host", ""));
+
+            let response = get.dispatch().await;
+            assert_eq!(response.status(), Status::BadRequest);
+        }
+
+        #[tokio::test]
+        async fn with_invalid_value_space_as_host() {
+            let (host_meta_cache, _host_meta_crawler) = crate::host_meta_crawling();
+            let infrastructure = Box::new(Dummy::new());
+            let apps = Arc::new(AppsService::new(Default::default(), infrastructure).unwrap());
+
+            let rocket = rocket::build()
+                .manage(host_meta_cache)
+                .manage(apps)
+                .mount("/", routes![crate::apps::routes::apps]);
+            let client = Client::tracked(rocket).await.expect("valid rocket");
+            let get = client
+                .get(rocket::uri!(crate::apps::routes::apps))
+                .header(Header::new("x-forwarded-host", " "));
+
+            let response = get.dispatch().await;
+            assert_eq!(response.status(), Status::BadRequest);
+        }
+
+        #[tokio::test]
+        async fn with_invalid_value_backslash_as_host() {
+            let (host_meta_cache, _host_meta_crawler) = crate::host_meta_crawling();
+            let infrastructure = Box::new(Dummy::new());
+            let apps = Arc::new(AppsService::new(Default::default(), infrastructure).unwrap());
+
+            let rocket = rocket::build()
+                .manage(host_meta_cache)
+                .manage(apps)
+                .mount("/", routes![crate::apps::routes::apps]);
+            let client = Client::tracked(rocket).await.expect("valid rocket");
+            let get = client
+                .get(rocket::uri!(crate::apps::routes::apps))
+                .header(Header::new("x-forwarded-host", "/"));
+
+            let response = get.dispatch().await;
+            assert_eq!(response.status(), Status::BadRequest);
+        }
+
+        #[tokio::test]
+        async fn with_invalid_proto() {
+            let (host_meta_cache, _host_meta_crawler) = crate::host_meta_crawling();
+            let infrastructure = Box::new(Dummy::new());
+            let apps = Arc::new(AppsService::new(Default::default(), infrastructure).unwrap());
+
+            let rocket = rocket::build()
+                .manage(host_meta_cache)
+                .manage(apps)
+                .mount("/", routes![crate::apps::routes::apps]);
+            let client = Client::tracked(rocket).await.expect("valid rocket");
+            let get = client
+                .get(rocket::uri!(crate::apps::routes::apps))
+                .header(Header::new("x-forwarded-proto", "."));
+
+            let response = get.dispatch().await;
+            assert_eq!(response.status(), Status::BadRequest);
+        }
+
+        #[tokio::test]
+        async fn with_valid_values() {
```

Comment:
This test tests nothing new, no new equivalence class… So the tests at the beginning should have you covered.